src/app/backend/resource/endpoint/endpoint.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 The Kubernetes Dashboard Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"github.com/kubernetes/dashboard/src/app/backend/api"
+	errorHandler "github.com/kubernetes/dashboard/src/app/backend/errors"
+	"github.com/kubernetes/dashboard/src/app/backend/resource/common"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	k8sClient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	kubeapi "k8s.io/kubernetes/pkg/api"
+	"strconv"
+)
+
+type Endpoint struct {
+	ObjectMeta api.ObjectMeta `json:"objectMeta"`
+	TypeMeta   api.TypeMeta   `json:"typeMeta"`
+
+	// Hostname, either as a domain name or IP address.
+	Host string `json:"host"`
+
+	// Name of the node the endpoint is located
+	NodeName *string `json:"nodeName"`
+
+	// Status of the endpoint
+	Ready string `json:"ready"`
+
+	// Array of endpoint ports
+	Ports []v1.EndpointPort `json:"ports"`
+
+	// List of non-critical errors, that occurred during resource retrieval.
+	Errors []error `json:"errors"`
+}
+
+// GetEndpointsForResource gets an Endpoint associated to resource with given name.
+func GetEndpointsForResource(client k8sClient.Interface, namespace, resourceName string) ([]v1.Endpoints, error) {
+	fieldSelector, err := fields.ParseSelector(kubeapi.EventInvolvedNameField + "=" + resourceName)
+	if err != nil {
+		return nil, err
+	}
+	channels := &common.ResourceChannels{
+		EndpointList: common.GetEndpointListChannelWithOptions(client,
+			common.NewSameNamespaceQuery(namespace),
+			metaV1.ListOptions{
+				LabelSelector: labels.Everything().String(),
+				FieldSelector: fieldSelector.String(),
+			},
+			1),
+	}
+
+	endpointsList := <-channels.EndpointList.List
+	if err := <-channels.EndpointList.Error; err != nil {
+		return nil, err
+	}
+
+	return endpointsList.Items, nil
+}
+
+// GetServicePods gets list of pods targeted by given label selector in given namespace.
+func GetServiceEndpoints(client k8sClient.Interface, namespace, name string) (*EndpointList, error) {
```

Comment:
You should test it with against authorization errors on service page. Turn off access to endpoints and service page should be still available. You can check how we are handling auth in our last pull requests.